spykes/popvis.py

```diff
@@ -0,0 +1,248 @@
+    visualizing firing activity of neuron populations
+
+    Parameters
+    ----------
+    neuron_list: list of NeuroVis objects
+
+    Internal variables
+    ------------------
+    neuron_list: list of NeuroVis objects
+    n_neurons: int
+
+    Callable methods
+    ----------------
+    get_psths
+    plot_heat_map
+
+    Class methods
+    -------------
+    _sort
+
+    """
+
+    def __init__(self, neuron_list):
+        """
+        Initialize the object
+        """
+        self.neuron_list = neuron_list
+        self.n_neurons = len(neuron_list)
+            
+    def get_psths(self, event=None, df=None, conditions=None,
+                 window=[-100, 500], binsize=10, event_name=None,
+                 conditions_names=None, plot=True, cmap=plt.cm.coolwarm):
+        """
+        Iterates through all neurons and computes their PSTH's
+
+        Parameters
+        ----------
+        event: str
+            Column/key name of DataFrame/dictionary "data" which contains
+            event times in milliseconds (e.g. stimulus/ trial/ fixation onset,
+            etc.)
+
+        df: DataFrame (or dictionary)
+
+        conditions: str
+            Column/key name of DataFrame/dictionary "data" which contains the
+            conditions by which the trials must be grouped
+
+        window: list of 2 elements
+            Time interval to consider (milliseconds)
+
+        binsize: int
+            Bin size in milliseconds
+
+        event_name: string
+            Legend name for event. Default is the actual 'event' name
+
+        conditions_names:
+            Legend names for conditions. Default are the unique values in
+            df['conditions']
+
+        plot: bool
+            Whether to automatically plot or not
+
+        cmap: str
+            Colormap for heatmap
+
+
+        Returns
+        -------
+        all_psth : dictionary
+            
+            With keys: 'event', 'conditions', 'binsize', 'window', and 'data'.
+            
+            Each entry in psth['data'] is itself a dictionary with keys of
+            each cond_id that correspond to the means for that condition
+        """
+
+        all_psth = dict()
+
+        all_psth['window'] = window
+        all_psth['binsize'] = binsize
+        all_psth['event'] = event
+        all_psth['conditions'] = conditions
+        all_psth['data'] = dict()
+
+        for i, neuron in enumerate(self.neuron_list):
+            
+            psth = neuron.get_psth(event=event, df=df, conditions=conditions,
+                                    window=window, binsize=binsize, plot=False)
+
+            for cond_id in np.sort(psth['data'].keys()):
+
+                if cond_id not in all_psth['data']:
+                    all_psth['data'][cond_id] = list()
+ 
+                all_psth['data'][cond_id].append(psth['data'][cond_id]['mean'])
+                
+                # TODO: what if cond_id empty for given neuron
```

Comment:
Should we generalize to whether it's not the same size or not? For example, in the notebook example, there are 200 bins. We should probably raise the error whenever there's not 200 bins